### PR TITLE
Eth2 client: improve init & fix trusted mode

### LINKED ISSUE #763: `Eth2OnNearClient`: renaming
Currently, `Eth2OnNearClient` has the same name as the first version: `EthClient`. It should be changed for differentiation to `Eth2Client`.

### LINKED ISSUE #765: `Eth2OnNearClient`: check configuration setup on Mainnet
`Eth2OnNearClient` should have a verification on init args that it can't be executed in the trustless mode without `bls` verification on Mainnet.

CC @mfornet

### LINKED ISSUE #766: `Eth2OnNearClient`: trusted mode fixes
When `Eth2OnNearClient` is in trusted mode, there's [verification](https://github.com/aurora-is-near/rainbow-bridge/blob/ebd777f8c32822942d275ec0e010a51acad385fc/contracts/near/eth2-client/src/lib.rs#L543) if a light-client update is allowed.
That method checks whether `signer_account_id` is equal to `trusted_signer`. However, as mentioned by @mfornet, there's no advantage to using `signer_account_id` over `predecessor_account_id`. Moreover, using `signer_account_id` could now allow using DAO as a trusted signer.

### PR DESCRIPTION
**Pull request by karim-en**
- Fix #765 
- Fix #766 
- Fix #763
- Additional unit tests

### CODE REVIEW

**Comment by sept-en** [comment]
@karim-en please rebase on top of `the-merge` branch

File: contracts/near/eth2-client/src/lib.rs [hunk 1]
```diff
@@ -85,6 +84,12 @@ impl EthClient {
                 args.validate_updates,
                 "The updates validation can't be disabled for mainnet"
             );
+
+            assert!(
+                (cfg!(feature = "bls") && args.verify_bls_signatures)
+                    || args.trusted_signer.is_some(),
+                "The client can't be executed in (trustless mode) and (without bls) on mainnet."
```
Comment (sept-en) [inline contracts/near/eth2-client/src/lib.rs:91]:
```suggestion
                "The client can't be executed in the trustless mode without BLS sigs verification on Mainnet"
```

File: contracts/near/eth2-client/src/tests/unit_tests.rs [hunk 2]
```diff
@@ -533,4 +533,37 @@ mod tests {
         update.sync_committee_update = None;
         contract.submit_beacon_chain_light_client_update(update);
     }
+
+    #[test]
+    #[should_panic(
+        expected = "The client can't be executed in (trustless mode) and (without bls) on mainnet."
```
Comment (sept-en) [inline contracts/near/eth2-client/src/tests/unit_tests.rs:539]:
```suggestion
        expected = "The client can't be executed in the trustless mode without BLS sigs verification on Mainnet"
```

File: contracts/near/eth2-client/src/tests/unit_tests.rs [hunk 3]
```diff
@@ -533,4 +533,37 @@ mod tests {
...
+            trusted_signer: None,
+        }));
+        init_input.network = "mainnet".to_string();
+        Eth2Client::init(init_input);
+    }
+
+    #[test]
+    #[cfg_attr(feature = "bls", ignore)]
+    #[should_panic(
+        expected = "The client can't be executed in (trustless mode) and (without bls) on mainnet."
```
Comment (sept-en) [inline contracts/near/eth2-client/src/tests/unit_tests.rs:556]:
```suggestion
        expected = "The client can't be executed in the trustless mode without BLS sigs verification on Mainnet"
```

File: contracts/near/eth2-client/src/tests/unit_tests.rs [hunk 4]
```diff
@@ -533,4 +533,37 @@ mod tests {
         update.sync_committee_update = None;
         contract.submit_beacon_chain_light_client_update(update);
     }
+
+    #[test]
+    #[should_panic(
+        expected = "The client can't be executed in (trustless mode) and (without bls) on mainnet."
+    )]
+    pub fn test_panic_on_init_in_trusted_mode_without_bls_on_mainnet() {
```
Comment (sept-en) [inline contracts/near/eth2-client/src/tests/unit_tests.rs:541]: I belive it's a typo
```suggestion
    )]
    pub fn test_panic_on_init_in_trustless_mode_without_bls_on_mainnet() {
```

File: contracts/near/eth2-client/src/tests/unit_tests.rs [hunk 5]
```diff
@@ -533,4 +533,37 @@ mod tests {
...
+        init_input.network = "mainnet".to_string();
+        Eth2Client::init(init_input);
+    }
+
+    #[test]
+    #[cfg_attr(feature = "bls", ignore)]
+    #[should_panic(
+        expected = "The client can't be executed in (trustless mode) and (without bls) on mainnet."
+    )]
+    pub fn test_panic_on_init_in_trusted_mode_without_bls_feature_flag() {
```
Comment (sept-en) [inline contracts/near/eth2-client/src/tests/unit_tests.rs:558]: I believe it's a typo
```suggestion
    pub fn test_panic_on_init_in_trustless_mode_without_bls_feature_flag() {
```

**Review by sept-en — CHANGES_REQUESTED** [review]

**Review by olga24912 — COMMENTED** [review]
OK

**Pull request merged**